Dockerfile

```diff
@@ -0,0 +1,63 @@
+# USAGE:
+# docker build --secret id=access_token,env=ACCESS_TOKEN --tag dsgrid --build-arg VERSION=x.y.z .
+# where ACCESS_TOKEN is a GitHub token that can clone the dsgrid repository
+
+# This container can be converted to a Singularity container on Eagle with these commands:
+# Acquire a compute node.
+# $ export SINGULARITY_TMPDIR=/tmp/scratch
+# $ module load singularity-container
+# Writable image for testing and development:
+# $ singularity build --sandbox dsgrid docker-archive://dsgrid_v0.1.0.tar
+# Read-only image for production:
+# $ singularity build dsgrid_v0.1.0.sif docker-archive://disco_v0.1.0.tar
+
+FROM python:3.8-slim
+
+ARG VERSION
+ARG SPARK_VERSION=3.2.0
+ARG HADOOP_VERSION=3.2
+ARG FULL_STR=spark-${SPARK_VERSION}-bin-hadoop${HADOOP_VERSION}
+ARG SPARK_HOME=/repos/${FULL_STR}
+
+RUN if [ -z "$VERSION" ]; then echo "VERSION must be specified"; exit 1; fi
+
+USER root
+
+RUN apt-get update \
+    && apt-get install -y jq git nano openjdk-11-jdk sysstat tmux tree vim wget zsh \
+    && rm -rf /var/lib/apt/lists/*
+
+RUN mkdir /data
+RUN mkdir /nopt
+RUN mkdir /projects
+RUN mkdir /repos
+RUN mkdir /scratch
+
+COPY docker/vimrc $HOME/.vimrc
+COPY docker/vimrc /data/vimrc
+
+RUN echo "$VERSION" > /repos/version.txt
+
+RUN --mount=type=secret,id=access_token ACCESS_TOKEN=$(cat /run/secrets/access_token) \
+    && git config --global url."https://${ACCESS_TOKEN}:@github.com/".insteadOf "https://github.com/"
+
+WORKDIR /repos
+# TODO: Find out how to install proper certificates.
```

Comment:
Does the buildstockbatch Dockerfile provide any clues about certificates:

```
FROM nrel/openstudio:2.9.1

ENV LANG=C.UTF-8 LC_ALL=C.UTF-8

RUN sudo apt update && \
    sudo apt install -y wget build-essential checkinstall libreadline-gplv2-dev libncursesw5-dev libssl-dev \
    libsqlite3-dev tk-dev libgdbm-dev libc6-dev libbz2-dev libffi-dev zlib1g-dev

RUN wget https://www.python.org/ftp/python/3.8.8/Python-3.8.8.tgz && \
    tar xzf Python-3.8.8.tgz && \
    cd Python-3.8.8 && \
    ./configure --enable-optimizations && \
    make altinstall && \
    rm -rf Python-3.8.8 && \
    rm -rf Python-3.8.8.tgz

RUN sudo apt install -y -V ca-certificates lsb-release && \
    wget https://apache.bintray.com/arrow/$(lsb_release --id --short | tr 'A-Z' 'a-z')/apache-arrow-archive-keyring-latest-$(lsb_release --codename --short).deb && \
    sudo apt install -y -V ./apache-arrow-archive-keyring-latest-$(lsb_release --codename --short).deb && \
    sudo apt update && \
    sudo apt install -y -V libarrow-dev libarrow-glib-dev libarrow-dataset-dev libparquet-dev libparquet-glib-dev

COPY . /buildstock-batch/
RUN python3.8 -m pip install /buildstock-batch
```

?

Reply:
Earlier I checked with Google and the answer was `ca-certifications`, which is above, but it didn't work. I'll explore a bit more.

Reply:
> Earlier I checked with Google and the answer was `ca-certificates`, which is above, but it didn't work. I'll explore a bit more.

The error was caused by being on the VPN.